Doc/library/shutil.rst

```diff
@@ -433,23 +433,42 @@ Directory and files operations
    When no *path* is specified, the results of :func:`os.environ` are used,
    returning either the "PATH" value or a fallback of :attr:`os.defpath`.
 
-   On Windows, the current directory is always prepended to the *path* whether
-   or not you use the default or provide your own, which is the behavior the
-   command shell uses when finding executables.  Additionally, when finding the
-   *cmd* in the *path*, the ``PATHEXT`` environment variable is checked.  For
-   example, if you call ``shutil.which("python")``, :func:`which` will search
-   ``PATHEXT`` to know that it should look for ``python.exe`` within the *path*
-   directories.  For example, on Windows::
+   On Windows, the current directory is prepended to the *path* if
+   the *mode* does not include ``os.X_OK``. When the *mode* does include
+   ``os.X_OK``, the Windows API ``NeedCurrentDirectoryForExePathW`` will be
+   consulted to determine if the current directory should be prepended to
+   *path*.
```

Comment:
```suggestion
   *path*. This will match the behavior of most shells on all platforms,
   preventing executables from being found in the current directory.
   For legacy shell behavior, pass *mode* without the ``os.X_OK`` flag.
```

Reply:
I don't think this is the case though. If the mode is executable, the curdir is added. Otherwise it asks `NeedCurrentDirectoryForExePath` if it should add curdir or not.

Reply:
Tried to clarify a bit.